cmd/id-exporter/main.go

```diff
@@ -24,15 +24,56 @@ type idExporter struct {
 	grace time.Duration
 }
 
-type id struct {
+// resultEntry is a JSON marshalable exporter result entry.
+type resultEntry struct {
+	// ID is exported to support marshaling to JSON.
 	ID int64 `json:"id"`
+
+	// Hostname is exported to support marshaling to JSON. Not all queries
+	// will fill this field, so it's JSON field tag marks at as
+	// omittable.
+	Hostname string `json:"hostname,omitempty"`
+}
+
+// reverseHostname converts (reversed) names sourced from the
+// registrations table to standard hostnames.
+func (r *resultEntry) reverseHostname() {
+	r.Hostname = sa.ReverseName(r.Hostname)
+}
+
+// idExporterResults is a container for a slice of pointers to
+// `resultEntry` objects. The `ids` field is passed as a selectable
+// 'holder' in calls to `dbMap.Select()`.
+type idExporterResults struct {
+	ids []*resultEntry
```

Comment:
Drive by, and I don’t really care, but in terms of the color of this shed, I generally like to avoid situations where the list is named like, ‘ids’ but the list contains objects that themselves have an ‘I’d’ but also other stuff. I’d rather it just be like ‘results’ or even ‘data’ so that it’s clearer from naming that there’s more there than only an id.

Reply:
Perhaps even better: I'm pretty sure this could just be `type idExporterResults []*resultEntry` instead of being a struct with a single field.

Reply:
Thank you, both of you for point this out!